src/native/external/zlib-ng/deflate_rle.c

```diff
@@ -58,7 +58,9 @@ Z_INTERNAL block_state deflate_rle(deflate_state *s, int flush) {
 
         /* Emit match if have run of STD_MIN_MATCH or longer, else emit literal */
         if (match_len >= STD_MIN_MATCH) {
-            check_match(s, s->strstart, s->strstart - 1, match_len);
+            Assert(s->strstart <= UINT16_MAX, "strstart should fit in uint16_t");
+            Assert(s->match_start <= UINT16_MAX, "match_start should fit in uint16_t");
+            check_match(s, (Pos)s->strstart, (Pos)s->strstart - 1, match_len);
```

Comment:
I don't want to restart the CI for this. I can either fix this in a separate PR, or we can just leave it as is:

- One assert is not needed
- The whole argument could be wrapped in parenthesis to convert the subtraction.


```suggestion
            Assert(s->strstart <= UINT16_MAX, "strstart-1 should fit in uint16_t");
            check_match(s, (Pos)s->strstart, (Pos)(s->strstart - 1), match_len);
```

I'll make sure to submit the upstream PR with this fix though.